core/common/src/main/java/alluxio/thread/AutopsyThread.java

```diff
@@ -18,7 +18,7 @@
  * This thread is able to capture uncaught exceptions from {@code run()}
  * so other classes can check the status of the thread and know why it crashed.
  */
-public class AutopsyThread extends Thread {
+public abstract class AutopsyThread extends Thread {
```

Comment:
I should probably let the AsyncJournalWriter.mFlushThread extend this logic too, that thread can also crash silently although we've never seen that (probably never noticed).

Reply:
a separate PR that is